File: Backend/DTOs/Product/ProductLongDto.cs
using Backend.DTOs.Review;

namespace Backend.DTOs.Product;

public record ProductLongDto : ProductShortDto
{
    public required string Description { get; init; }
    public required int Calories { get; init; }
    public required string Category { get; init; }
    public required List<ReviewDto> Reviews { get; init; }
    public required List<string> Ingredients { get; init; }
}

File: frontend/src/utils/types/product/ProductLongDto.ts
import type { ProductShortDto } from '@/utils/types/product/ProductShortDto';
import type { ReviewDto } from '@/utils/types/review/ReviewDto';

export type ProductLongDto = ProductShortDto & {
  calories: number;
  categoryName: string;
  reviews: ReviewDto[];
  ingredients: string[];
};
